packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -133,71 +122,104 @@ export class RemoteTableService {
     input: RemoteTableInput,
     remoteServer: RemoteServerEntity<RemoteServerType>,
     workspaceId: string,
-    workspaceDataSource: DataSource,
-    dataSourceMetadata: DataSourceEntity,
   ) {
-    const localSchema = dataSourceMetadata.schema;
+    if (!input.schema) {
+      throw new Error('Schema is required for syncing remote table');
+    }
 
-    // TODO: Add strong typing for remote table columns. Will be done when we have another use case than Postgres
     const remoteTableColumns = await this.fetchTableColumnsSchema(
       remoteServer,
       input.name,
       input.schema,
     );
 
-    const foreignTableColumns = remoteTableColumns
-      .map((column) => `"${column.column_name}" ${column.data_type}`)
-      .join(', ');
-
-    const remoteTableName = `${camelCase(input.name)}Remote`;
+    const remoteTableName = getRemoteTableName(input.name);
     const remoteTableLabel = camelToTitleCase(remoteTableName);
 
     // We only support remote tables with an id column for now.
     const remoteTableIdColumn = remoteTableColumns.filter(
-      (column) => column.column_name === 'id',
+      (column) => column.columnName === 'id',
     )?.[0];
 
     if (!remoteTableIdColumn) {
       throw new Error('Remote table must have an id column');
     }
 
-    await workspaceDataSource.query(
-      `CREATE FOREIGN TABLE ${localSchema}."${remoteTableName}" (${foreignTableColumns}) SERVER "${remoteServer.foreignDataWrapperId}" OPTIONS (schema_name '${input.schema}', table_name '${input.name}')`,
+    const dataSourcesMetatada =
+      await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
+        workspaceId,
+      );
+
+    if (!dataSourcesMetatada) {
+      throw new NotFoundException('Workspace data source does not exist');
+    }
+
+    this.workspaceMigrationService.createCustomMigration(
+      generateMigrationName(`create-remote-table-${remoteTableName}`),
+      workspaceId,
+      [
+        {
+          name: remoteTableName,
+          action: 'create_foreign_table',
+          foreignTable: {
+            columns: remoteTableColumns.map(
+              (column) =>
+                ({
+                  columnName: column.columnName,
+                  columnType: column.dataType,
+                }) satisfies WorkspaceMigrationColumnDefinition,
+            ),
+            referencedTableName: input.name,
+            referencedTableSchema: input.schema,
+            foreignDataWrapperId: remoteServer.foreignDataWrapperId,
+          } satisfies WorkspaceMigrationForeignTable,
+        },
+        {
+          name: remoteTableName,
+          action: 'alter',
+          columns: [
+            {
+              action: WorkspaceMigrationColumnActionType.CREATE_COMMENT,
```

Comment:
@Weiko do you think that should live in the create_foreign_table migration? That's what we did for a classic create_table migration since the comment is always written on the table